doc/man/sphinx-apidoc.rst

```diff
@@ -126,29 +126,260 @@ These options are used when :option:`--full` is specified:
+
+.. data:: show_headings
+
+True unless :option:`--no-headings` was given
+
+
+The default ``package.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/package.rst_t
+   :language: jinja
+
+
+The module template
+~~~~~~~~~~~~~~~~~~~
+
+The module template is used to render modules. This happens only if
+:option:`--separate` was given, or if *MODULE_PATH* contains standalone
+modules instead of packages.
+
+It uses the :data:`automodule_options` and :data:`show_headings` variables of
+``package.rst_t``, and additionally:
+
+.. data:: qualname
+
+The fully qualified name of the module.
+
+.. data:: basename
+
+An alias for :data:`qualname`.
+
+The default ``module.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/module.rst_t
+   :language: jinja
+
+
+Accessing module members in templates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.3
+   Advanced templating with ``get_members``
+
+Both the ``package.rst_t`` and ``module.rst_t`` templates have access to a
+:func:`get_members` function that allows to extract the members of the package or
+module being rendered:
+
+
+.. py:function:: get_members(fullname, typ=None, include_imported=False, out_format='names', in_list=None, include_private=*private*, known_refs=None)
+
+   Return a list of members.
+
+   :param str fullname:
+      The fully qualified name of the module for which to get the members
+
+   :param Optional[str] typ:
+      One of None, 'function', 'class', 'exception', 'data'. If None, return
+      members of any type. Otherwise return only members of the given `typ`
+
+   :param bool include_imported:
+      If True, include members that are imported from other modules. If False,
+      only return members that are defined directly in the module
+
+   :param str out_format:
+      One of 'names', 'fullnames', 'refs', and 'table'
+
+   :param Optional[Union[str,Tuple[str]] in_list:
+      If not None, name or tuple of names of module attribute(s) (e.g.
+      ``'__all__'`` or ``('__all__', '__private__')``). Only members whose
+      names appears in the list(s) will be returned.
+
+   :param bool include_private:
+      If True, include members whose names starts with an underscore. Defaults
+      to False unless :option:`--private` is given.
+
+   :param Optional[Union[dict,str]] known_refs:
+      If not None, a mapping of names to rull rst-formatted references. If
+      given as a str, the mapping will be taken from the module attribute of
+      the given name. This is used only in conjunction with
+      ``out_format=refs``, to override automatically detected reference
+      location, or to provide references for object that cannot be located
+      automatically. The most common example for this is data members that are
+      exported by a package but are defined in a sub-module.
+
+   :return:
+      a list of strings, depending on `out_format`.
+
+      * If ``out_format='names'`` (default), the simple names of all qualifying
```

Comment:
If we remove the `out_format` option from this entirely, then that simplifies the changes greatly.

> I feel this is too complex. It seems the responsibility of this function is listing attributes, filtering, **formatting** and so on

That addresses removing the formatting concern.